lib/europeana/api/record/hierarchy.rb

```diff
@@ -1,59 +1,93 @@
-require 'rubytree'
-require 'active_support/core_ext/object/blank'
-
 module Europeana
   module API
     class Record
       ##
       # Retrieve record hierarchies over the Europeana API
-      class Hierarchy < Tree::TreeNode
+      class Hierarchy < HashWithIndifferentAccess
         include Requestable
 
-        attr_reader :record
+        attr_reader :record_id, :params
 
         ##
         # @param record [Europeana::API::Record]
-        def initialize(record)
-          @record = record
-          super(record.id, execute_request[:self])
-          fetch_family
-        end
+        def initialize(record_id_or_hash, params = {})
```

Comment:
Unused method argument - `params`. If it's necessary, use `_` or `_params` as an argument name to indicate that it won't be used.<br>Method has too many lines. [11/10]